corehq/apps/userreports/reports/specs.py

```diff
@@ -341,20 +333,51 @@ def get_query_column_ids(self):
         return [self.column_id]
 
 
-class ConditionalAggregationColumn(_CaseExpressionColumn):
+class ConditionalAggregationRangesColumn(_CaseExpressionColumn):
     """Used for grouping by SQL conditionals"""
     type = TypeProperty('conditional_aggregation')
     _agg_column_type = ConditionalAggregation
+    field = StringProperty(required=True)
+    ranges = DictProperty()
 
     def get_whens(self):
-        return {k: bindparam(None, v) for k, v in self.whens.items()}
+        whens = {}
+        for value, bounds in self.ranges.items():
+            if len(bounds) != 2:
+                raise BadSpecError('Range must contain 2 items, contains {}'.format(len(bounds)))
+            try:
+                bounds = [float(b) for b in bounds]
```

Comment:
why is this converting to `float`? Are there going to be strange edge cases where someone specifies `2`, but its actually `2.00000000003` or `1.999999999999997` which will make a big difference depending on if these are inclusive/exclusive

Reply:
The intent was to allow for decimals. But the existing reports all use ints, so switching this out for an int cast seems fine - 7dce64b70378f8121eeb7b65281d44eb50ac5623